### PR TITLE
fix l1 merge logic

### PR DESCRIPTION
**Pull request by rymarczy**
update `l1_full_trip_events.py` to perform all merge logic with sql

remove use of `.in` and `.notin` and replace with join logic

Pandas merge operations do not work with byte objects being used for record hashes. This update performs that merge logic in SQL to avoid pandas issues.

Asana Task: https://app.asana.com/0/1202841571879768/1202986972911822

### CODE REVIEW

File: performance_manager/lib/postgres_schema.py [hunk 1]
```diff
@@ -164,16 +164,19 @@ class FullTripEvents(SqlBase):  # pylint: disable=too-few-public-methods
...
         sa.Integer,
         sa.ForeignKey("eventsVehiclePositions.pk_id", ondelete="CASCADE"),
         nullable=True,
+        index=True,
     )
     fk_tu_stopped_event = sa.Column(
         sa.Integer,
         sa.ForeignKey("eventsTripUpdates.pk_id", ondelete="CASCADE"),
         nullable=True,
+        index=True,
```
Comment (rymarczy) [inline performance_manager/lib/postgres_schema.py:179]: add index on ForeignKey fields

File: performance_manager/lib/logging_utils.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def log_start(self) -> None:
 
     def log_complete(self) -> None:
         """log the completion of a proccess with duration"""
-        duration = self.start_time - time.time()
+        duration = time.time() - self.start_time
```
Comment (rymarczy) [inline performance_manager/lib/logging_utils.py:49]: get rid of those negative durations

Comment (mzappitello) [reply performance_manager/lib/logging_utils.py:49]: ha. i had this in my debugging branch as well.

File: performance_manager/lib/postgres_schema.py [hunk 3]
```diff
@@ -11,7 +11,7 @@ class VehiclePositionEvents(SqlBase):  # pylint: disable=too-few-public-methods
 
     __tablename__ = "eventsVehiclePositions"
 
-    pk_id = sa.Column(sa.Integer, primary_key=True)
+    pk_id = sa.Column(sa.Integer, primary_key=True, index=True)
```
Comment (mzappitello) [inline performance_manager/lib/postgres_schema.py:14]: what does `index=True` do here?

Comment (rymarczy) [reply performance_manager/lib/postgres_schema.py:14]: It tells the database to create an index on the Column, I kind of assumed that the `primary_key` flag would force an index creation, but I didn't see an index for these columns in pgadmin when I was looking at the tables.

File: performance_manager/lib/l1_full_trip_events.py [hunk 4]
```diff
@@ -16,208 +13,230 @@
...
-    ]
-
-    process_logger = ProcessLogger("l1_pull_events")
-    process_logger.log_start()
-
-    for event_type in events_list.values():
-        # get dataframe from L0 table select
-        event_type.merge_dataframe = db_manager.select_as_dataframe(
-            event_type.select_query
+    merged_query = (
```
Comment (mzappitello) [inline performance_manager/lib/l1_full_trip_events.py:156]: this query is magic to me. i'm having a bit of trouble parsing it all out.

File: performance_manager/lib/l1_full_trip_events.py [hunk 5]
```diff
@@ -16,208 +13,230 @@
...
-    # events, even if they do not have associated moving event
-    return_dataframe = return_dataframe.merge(
-        events_list["stopped_vp"].merge_dataframe, how="outer", on="hash"
-    )
+    # if data is retrieved from L0 table, ensure number columns are
+    # Int64 type to ensure consistent hashing
+    for column in convert_to_number_columns:
+        merged_dataframe[column] = pandas.to_numeric(
+            merged_dataframe[column]
+        ).astype("Int64")
```
Comment (mzappitello) [inline performance_manager/lib/l1_full_trip_events.py:228]: kind of surprising that this isn't a built in function.

File: performance_manager/lib/l1_full_trip_events.py [hunk 6]
```diff
@@ -16,208 +13,230 @@
...
+        .join(
+            stop_tu_cte,
+            sa.and_(
+                move_vp_cte.c.stop_sequence == stop_tu_cte.c.stop_sequence,
+                move_vp_cte.c.stop_id == stop_tu_cte.c.stop_id,
+                move_vp_cte.c.direction_id == stop_tu_cte.c.direction_id,
+                move_vp_cte.c.route_id == stop_tu_cte.c.route_id,
+                move_vp_cte.c.start_date == stop_tu_cte.c.start_date,
+                move_vp_cte.c.start_time == stop_tu_cte.c.start_time,
+                move_vp_cte.c.vehicle_id == stop_tu_cte.c.vehicle_id,
```
Comment (mzappitello) [inline performance_manager/lib/l1_full_trip_events.py:194]: is this section doing the work that the hash was doing before?

ensuring that all this data is the same?

**Pull request merged**